app/controllers/comments_controller.rb

```diff
@@ -1,13 +1,42 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
+
 def create
   @post = Post.find(params[:post_id])
   @post.comments.create(comment_params.merge(:user => current_user))
   redirect_to post_path(@post)
 end
 
+def edit
+  @comment = Comment.find(params[:comment_id])
+end
+
+def update
+  @comment = Comment.find(params[:comment_id])
+  @comment.update_attributes(comment_params)
+  redirect_to post_path(@post)
```

Comment:
You never set the `@post` variable.... you'd need to, to use the variable.  Alternatively you could just do:

```
redirect_to post_path(@comment.post)
```